plugin/schema.go

```diff
@@ -0,0 +1,82 @@
+package plugin
+
+import (
+	"reflect"
+	"strings"
+)
+
+// EntrySchema represents an entry's schema.
+type EntrySchema struct {
+	Type      string        `json:"type"`
+	ShortType string        `json:"short_type"`
+	Singleton bool          `json:"singleton"`
+	Actions   []string      `json:"actions"`
+	Children  []EntrySchema `json:"children"`
+	entry     Entry
+}
+
+// ChildSchemas is a helper that's used to implement Parent#ChildSchemas
+func ChildSchemas(childTemplates ...Entry) []EntrySchema {
```

Comment:
I'm not sure Entry is the right thing to include here if we have to change NewEntry to satisfy it. I'd be fine with a narrower interface.

Reply:
Think this was addressed in https://github.com/puppetlabs/wash/pull/321#discussion_r292613928

I guess I could do a type-alias `type EntryTemplate = Entry`, but that's an extra thing people have to remember. Then again, it might make the method documentation clearer (and explicitly indicate that it's separate from `plugin.Schema`).